test/framework/resources/templates/reserve-capacity.yaml.tpl

```diff
@@ -31,12 +41,5 @@ spec:
       - name: pause-container
         image: gcr.io/google_containers/pause-amd64:3.1
         imagePullPolicy: IfNotPresent
-        resources:
```

Comment:
Just to mention, should not be a concern in this specific test: the scale-down of the autoscaler relies on the aggregated resource-requests of the pods running in the machine. So, with this approach, the node should be a candidate for the scaledown immediately after the `scaleDownDelayAfterAdd ` period, irrespective of the workload running in the machine.

For `scale from/to zero test`, a separate `testWorkerPool` is created with certain taints only tolerated by the `reserver-capacity` pods, maybe this could be a way fwd for more advanced tests in the future.

Reply:
Are you saying, that clusterautoscaler will try to remove the newly added node although there is a pod running there with required anti-affinity?

Reply:
Hm, good point, it should ideally not, else the pod will be unschedulable. I guess, it shouldn't be the concern then.